extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseDialog.ts

```diff
@@ -54,6 +54,19 @@ export class CreateProjectFromDatabaseDialog {
 		await this.initDialogComplete.promise;
 
 		if (this.profile) {
+			const connections = await azdataType.connection.getConnections(true);
+			let connection = connections.find(c => c.connectionId === this.profile!.id);
+
+			if (!connection) {
+				// if the connection clicked on isn't currently connected, try to connect
+				const result = await azdataType.connection.connect(this.profile, true, false);
+
+				if (!result.connected) {
+					// if can't connect automatically, open connection dialog with the info from the profile
+					await azdataType.connection.openConnectionDialog(undefined, this.profile);
```

Comment:
Not sure how I feel about this; gif looks a little jarring to open one dialog, pause for a moment, then open another one, losing some context along the way.

I think it'd feel smoother if did nothing if it couldn't connect, or maybe gave it an error box around the field in the dialog saying it couldn't connect.

Reply:
Yeah makes sense that it was a little jarring opening the connection dialog after opening the create project dialog. Does opening the connection dialog (if auto connecting didn't work) first look better?
![openConnDialogFirst](https://user-images.githubusercontent.com/31145923/163857038-30f3ab6f-a040-43c9-bd61-c809737c7214.gif)

If not, I can remove opening the connection dialog at all if the connection couldn't automatically be made, but then the user would have to open the connection dialog themselves anyway if they wanted to create a project from a database on this connection.